internal/vendors/equinix/equinix.go

```diff
@@ -0,0 +1,204 @@
+	)
+	tc := oauth2.NewClient(ctx, ts)
+
+	ghClient := github.NewClient(tc)
+
+	// parse S3 endpoint and bucket from cfgSyncer.RepositoryURL
+	s3DstEndpoint, s3DstBucket, err := config.ParseRepositoryURL(cfgSyncer.RepositoryURL)
+	if err != nil {
+		return nil, err
+	}
+
+	dstS3Config := &config.S3Bucket{
+		Region:    cfgSyncer.RepositoryRegion,
+		Endpoint:  s3DstEndpoint,
+		Bucket:    s3DstBucket,
+		AccessKey: os.Getenv("S3_ACCESS_KEY"),
+		SecretKey: os.Getenv("S3_SECRET_KEY"),
+	}
+
+	// init inventory
+	i, err := inventory.New(ctx, cfgSyncer.ServerServiceURL, cfgSyncer.ArtifactsURL, logger)
+	if err != nil {
+		return nil, err
+	}
+
+	// init rclone filesystems for tmp and dst files
+	vendors.SetRcloneLogging(logger)
+
+	dstFs, err := vendors.InitS3Fs(ctx, dstS3Config, "/")
+	if err != nil {
+		return nil, err
+	}
+
+	tmpFs, err := vendors.InitLocalFs(ctx, &vendors.LocalFsConfig{Root: "/tmp"})
+	if err != nil {
+		return nil, err
+	}
+
+	return &Equinix{
+		firmwares: firmwares,
+		logger:    logger,
+		metrics:   vendors.NewMetrics(),
+		inventory: i,
+		ghClient:  ghClient,
+		dstCfg:    dstS3Config,
+		dstFs:     dstFs,
+		tmpFs:     tmpFs,
+	}, nil
+}
+
+func (e *Equinix) Stats() *vendors.Metrics {
+	return e.metrics
+}
+
+func (e *Equinix) Sync(ctx context.Context) error {
+	for _, fw := range e.firmwares {
+		// In case the file already exists in dst, don't copy it
+		if exists, _ := fs.FileExists(ctx, e.dstFs, vendors.DstPath(fw)); exists {
+			e.logger.WithFields(
+				logrus.Fields{
+					"filename": fw.Filename,
+				},
+			).Debug("firmware already exists at dst")
+
+			continue
+		}
+
+		err := e.getFileFromGithub(ctx, fw)
+		if err != nil {
+			return err
+		}
+
+		// Verify file checksum
+		tmpFilename := e.tmpFs.Root() + "/" + fw.Filename
+		if !vendors.ValidateChecksum(tmpFilename, fw.Checksum) {
+			return errors.Wrap(vendors.ErrChecksumValidate, fmt.Sprintf("tmpFilename: %s, expected checksum: %s", tmpFilename, fw.Checksum))
+		}
+
+		e.logger.WithFields(
+			logrus.Fields{
+				"src": fw.UpstreamURL,
+				"dst": vendors.DstPath(fw),
+			},
+		).Info("sync Equinix")
+
+		// Copy from tmpfs to dstfs
+		err = operations.CopyFile(ctx, e.dstFs, e.tmpFs, vendors.DstPath(fw), fw.Filename)
+		if err != nil {
+			return err
+		}
+
+		err = e.inventory.Publish(ctx, fw, vendors.DstPath(fw))
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (e *Equinix) getFileFromGithub(ctx context.Context, fw *serverservice.ComponentFirmwareVersion) error {
+	owner, repo, tag, filename, err := parseGithubReleaseURL(fw.UpstreamURL)
+	if err != nil {
+		return err
+	}
+
+	release, _, err := e.ghClient.Repositories.GetReleaseByTag(ctx, owner, repo, tag)
+	if err != nil {
+		return err
+	}
+
+	asset, err := getAssetByName(filename, release.Assets)
+	if err != nil {
+		return err
+	}
+
+	// Give enough time for the client to download the binary file.
+	redirectClient := &http.Client{
+		Timeout: time.Second * 300,
```

Comment:
would be ideal to define this timeout  value as a const

Reply:
Done in d21240a